tasks/16.txt

```diff
@@ -1,3 +1,8 @@
 Create a SQL query to return all orders where ship postal code is provided:
 
-| OrderId | CustomerId | ShipCountry |
\ No newline at end of file
+| OrderId | CustomerId | ShipCountry |
+
+db.orders.aggregate([
+{$match: {ShipPostalCode: {$not: {$type: 10},$exists: true}}},
+{$project: {OrderID: 1, CustomerID: 1, ShipCountry: 1}}
+]);
```

Comment:
1. For $type more friendly to use string aliases for types. That is not really affecting to performance
2. why not just $ne: null instead of $not: {$type: 10}
3. please fix orders also db.orders.updateMany({ShipPostalCode:'NULL'},{$set:{ShipPostalCode:null}})